dirs/dirs.go

```diff
@@ -138,6 +138,9 @@ var (
 	SysfsDir        string
 
 	FeaturesDir string
+
+	// Comma separated list of user defined home directories in system-params
+	SnapHomeDirs string
```

Comment:
while I think that in most cases it will be accessed and changed holding the state global lock is hard to be sure, the safer approach is to have a mutex around this:

```
var (
    snapHomeDirsMu sync.Mutex
    snapHomdeDirs string
)

func SnapHomeDirs() string {
    snapHomeDirsMu.Lock()
    defer snapHomeDirsMu.Unlock()
    return snapHomeDirs
}

func SetSnapHomeDirs(...) {
...
}
```

Also here maybe it makes sense to make it directly a []string and not have the users split the content by themselves